packages/x-components/src/components/__tests__/preselected-filters.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { mount } from '@vue/test-utils';
+import { createRawFilters } from '../../utils/filters';
+import { baseSnippetConfig } from '../../views/base-config';
+import PreselectedFilters from '../preselected-filters.vue';
+
+function renderPreselectedFilters({
+  filters,
+  snippetFilters
+}: RenderPreselectedFiltersOptions = {}): RenderPreselectedFiltersAPI {
+  const emit = jest.fn();
+  mount(PreselectedFilters, {
+    provide: {
+      snippetConfig: { ...baseSnippetConfig, filters: snippetFilters }
+    },
+    propsData: { filters },
+    mocks: {
+      $x: {
+        emit
+      }
+    }
+  });
+
+  return {
+    emit
+  };
+}
+
+describe('testing Preselected filters component', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('does not emit the event when neither filters nor snippet config filters are provided', () => {
+    const { emit } = renderPreselectedFilters();
+
+    expect(emit).not.toHaveBeenCalled();
+  });
+  it('emits the event when filters are provided in the snippet config', () => {
+    const snippetFilters = [
```

Comment:
Move this to a constant outside of the tests and use that instead of repeating the same filters in the other tests. Also, rename it to `filters` or something less specific to the snippet and then use it with `snippetFilters: filtersConst`